pkg/utils/status_test.go

```diff
@@ -0,0 +1,1614 @@
+			if (err != nil) != test.wantErr {
+				t.Errorf("HandleValidationError() error = %v, wantErr %v", err, test.wantErr)
+				return
+			}
+			assert.Equal(t, test.expectedResult, result)
+			assert.Equal(t, constants.Failed, test.args.instance.GetCSMStatus().State)
+		})
+	}
+}
+
+func TestHandleSuccess(t *testing.T) {
+	type args struct {
+		ctx       context.Context
+		instance  *csmv1.ContainerStorageModule
+		r         ReconcileCSM
+		oldStatus *csmv1.ContainerStorageModuleStatus
+		newStatus *csmv1.ContainerStorageModuleStatus
+	}
+
+	tests := []struct {
+		name string
+		args args
+		want reconcile.Result
+	}{
+		{
+			name: "Test TestHandleSuccess with no change in status",
+			args: args{
+				ctx:      context.Background(),
+				instance: createCSM("powerflex", "powerflex", csmv1.PowerFlex, csmv1.Replication, true, nil),
+				r: &FakeReconcileCSM{
+					Client:    ctrlClientFake.NewClientBuilder().Build(),
+					K8sClient: fake.NewSimpleClientset(),
+				},
+				oldStatus: &csmv1.ContainerStorageModuleStatus{
+					ControllerStatus: csmv1.PodStatus{
+						Available: "1",
+						Failed:    "0",
+						Desired:   "1",
+					},
+					NodeStatus: csmv1.PodStatus{
+						Available: "1",
+						Failed:    "0",
+						Desired:   "1",
+					},
+					State: constants.Succeeded,
+				},
+				newStatus: &csmv1.ContainerStorageModuleStatus{
+					ControllerStatus: csmv1.PodStatus{
+						Available: "1",
+						Failed:    "0",
+						Desired:   "1",
+					},
+					NodeStatus: csmv1.PodStatus{
+						Available: "1",
+						Failed:    "0",
+						Desired:   "1",
+					},
+					State: constants.Succeeded,
+				},
+			},
+			want: reconcile.Result{
+				Requeue: false,
+			},
+		},
+		// { This needs to be fixed
+		// 	name: "Test TestHandleSuccess with change in status",
+		// 	args: args{
+		// 		ctx:      context.Background(),
+		// 		instance: createCSM("powerflex", "powerflex", csmv1.PowerFlex, csmv1.Replication, true, nil),
+		// 		r: &FakeReconcileCSM{
+		// 			Client:    ctrlClientFake.NewClientBuilder().Build(),
+		// 			K8sClient: fake.NewSimpleClientset(),
+		// 		},
+		// 		oldStatus: &csmv1.ContainerStorageModuleStatus{
+		// 			ControllerStatus: csmv1.PodStatus{
+		// 				Available: "0",
+		// 				Failed:    "0",
+		// 				Desired:   "1",
+		// 			},
+		// 			NodeStatus: csmv1.PodStatus{
+		// 				Available: "0",
+		// 				Failed:    "0",
+		// 				Desired:   "1",
+		// 			},
+		// 			State: constants.Succeeded,
+		// 		},
+		// 		newStatus: &csmv1.ContainerStorageModuleStatus{
+		// 			ControllerStatus: csmv1.PodStatus{
+		// 				Available: "1",
+		// 				Failed:    "0",
+		// 				Desired:   "1",
+		// 			},
+		// 			NodeStatus: csmv1.PodStatus{
+		// 				Available: "1",
+		// 				Failed:    "0",
+		// 				Desired:   "1",
+		// 			},
+		// 			State: constants.Succeeded,
+		// 		},
+		// 	},
+		// 	want: reconcile.Result{
+		// 		Requeue: true,
+		// 	},
+		// },
```

Comment:
Should we remove this test, or do we intend to fix it?

Reply:
This test has been fixed. The commented code is now uncommented. Resolving this item.